diff --git a/patch/prebuild_eudev/rules.d/50-udev-default.rules b/patch/prebuild_eudev/rules.d/50-udev-default.rules
index a64fbd405da34b39d43db210b7e8e70669cee7c5..04c149f8e65697564de6c2f35950414f62689c4f 100644
--- a/patch/prebuild_eudev/rules.d/50-udev-default.rules
+++ b/patch/prebuild_eudev/rules.d/50-udev-default.rules
@@ -88,4 +88,7 @@ KERNEL=="udmabuf", GROUP="kvm"
 
 SUBSYSTEM=="ptp", ATTR{clock_name}=="KVM virtual PTP", SYMLINK += "ptp_kvm"
 
+# When plug-in MEIG modem, add read and write permissions for riladapter_host.
+SUBSYSTEM=="tty", KERNEL=="ttyUSB[0-4]*", GROUP="tty", MODE="0666"
+
 LABEL="default_end"
